src/workload.rs

```diff
@@ -366,7 +366,9 @@ pub struct WorkloadStore {
     workloads: HashMap<IpAddr, Workload>,
     // workload_to_vip maintains a mapping of workload IP to VIP. This is used only to handle removals.
     workload_to_vip: HashMap<IpAddr, HashSet<(SocketAddr, u16)>>,
-    vips: HashMap<SocketAddr, HashSet<Upstream>>,
+    // vips maintains a mapping of socket address with service port to workload ip and socket address
+    // with target ports in hashset.
+    vips: HashMap<SocketAddr, HashSet<(u16, IpAddr)>>,
```

Comment:
to be consistent with workload_to_vip, maybe HashSet<(IpAddr, u16)

Reply:
okay, it makes sense

Reply:
@hzxuzhonghu done!